metricbeat/module/vsphere/datastorecluster/datastorecluster_test.go

```diff
@@ -0,0 +1,90 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package datastorecluster
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/vmware/govmomi/simulator"
+
+	mbtest "github.com/elastic/beats/v7/metricbeat/mb/testing"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+func TestFetchEventContents(t *testing.T) {
+	model := simulator.VPX()
+	err := model.Create()
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() { model.Remove() })
```

Comment:
Looking at the signature of `model.Remove` method [here](https://github.com/vmware/govmomi/blob/7be6ec95bb835175d03ac5689cad7c757117a301/simulator/model.go#L861a) I think we could simply pass in model.Remove like so:
```suggestion
	t.Cleanup(model.Remove)
```